### ISSUE TITLE
False positive error "NO DATA: Low on available disk space"

### ISSUE THREAD

**Issue by rikukissa** [opencrvs/opencrvs-core#7149]
This error is still unresolved and can cause a lot of confusion

**rikuland-staging.opencrvs.org NO DATA: Low on available disk space 🚨**
> Reason: Max of system.filesystem.used.pct has reported no data over the past 1h for metricbeat-rikuland-staging. Login to https://kibana.rikuland-staging.opencrvs.org to view the alert.

**Comment by euanmillar** [comment]
False positives especially on such urgent matter can be dangerous so we decided to fix this asap

**Comment by makelicious** [comment]
False positive error "NO DATA: Low on available disk space"


TL:DR;
Update to next major.

Every n seconds metricbeat runs into following error:
```
ERROR	metrics/metrics.go:376	error getting cgroup stats: error fetching stats for controller io: error fetching IO stats: error getting io.stats for path /hostfs/sys/fs/cgroup: error scanning file: /hostfs/sys/fs/cgroup/io.stat: unexpected EOF
```

Error is caused by `io.stat` file format which depends on kernel version.
formatting rules are determined here: https://github.com/elastic/elastic-agent-system-metrics/blob/v0.4.4/metric/system/cgroup/cgv2/io.go#L80

Relevant issue is fixed here: https://github.com/elastic/elastic-agent-system-metrics/pull/77 ((available since `elastic-agent-system-metrics` `v0.5`)

`elastic-agent-system-metrics` is a dependency of beats. Fix has not been backported to `v7.17.x`
The fixed version is available on beats `v8.8` onwards.

General:
1. Metricbeat gathers metrics.
2. Metricbeat consists of modules (e.g. 'system') Modules consists of metricsets. (e.g. 'filesystem').
3. Alerts are configured in `config.ndjson`

"Low on available disk space"
4. Alert checks a metric `system.filesystem.used.pct`
5. Alert check is run once per hour (1h interval).
6. When data is not available `"alertOnNoData": true`, alert is triggered
7. Elastalert checks alerts every 5 minutes `timeframe: minutes: 5` (`/opencrvs-countryconfig/infrastructure/monitoring/elastalert/rules/alert.yaml`)
8. When alert keyword matches, an email is send.
9. Every n seconds metricbeat runs into following error:
    ```
    ERROR	metrics/metrics.go:376	error getting cgroup stats: error fetching stats for controller io: error fetching IO stats: error getting io.stats for path /hostfs/sys/fs/cgroup: error scanning file: /hostfs/sys/fs/cgroup/io.stat: unexpected EOF
    ```

Since `io.stat` does not seem to be immediately related to `system.filesystem.used.pct`

10. Removed other system modules from metricbeat

```
metricbeat.modules:
  #------------------------------- System Module -------------------------------
  - module: system
    period: 10s
    metricsets:
      - filesystem
    processors:
      - drop_event.when.regexp:
          system.filesystem.mount_point: '^/(sys|cgroup|proc|dev|etc|host|lib)($|/)'
```

11. Error still reoccurs after deployment.
12. debug log sends following event after the error

```
{
    ...rest of the payload,
  "metrics": {
    "metricbeat": {
      "system": {
        "filesystem": {
          "failures": 0,
          "events": 0,
          "success": 0
        }
      }
    }
  }
}
```

13. Assuming the filter query in alert is searching against this and similar events, it should report no data.
`"filterQuery": "{\"bool\":{\"should\":[{\"match_phrase\":{\"system.filesystem.device_name\":\"/dev/mapper/cryptfs\"}}],\"minimum_should_match\":1}}"`

Next steps
a) Update to next major and see if the error disappears
b) Investigate further `config.ndjson` alert configuration for `Low on available disk space`. Is there something missing

**Comment by SyedaAfrida** [comment]
Cannot be tested from QA environment
